source/agora/test/Network.d

```diff
@@ -86,3 +86,19 @@ unittest
     auto network = makeTestNetwork!Network(NetworkTopology.Simple, 4);
     network.waitUntilConnected(4);
 }
+
+///
+unittest
+{
+    import std.digest.sha; 
+    auto network = makeTestNetwork!Network(NetworkTopology.Simple, 4);
+    network.waitUntilConnected(4);
+
+    assert (network.apis.keys.length == 4);
+
+    Hash h1 = sha256Of("Message No. 1");
+    auto key = network.apis.keys[0];
+    auto n1 = network.apis[key];
+
+    n1.setMessage(h1);
```

Comment:
Looks like the best way to test this would be to add an endpoint to check if a message exists. Something like `bool hasMessage (Hash msg)` in `agora.common.API : API`. What do you think @MukeunKim @AndrejMitrovic ?

Reply:
That's a good idea.

Reply:
Can I do this? bool hasMessage (Hash msg)

Reply:
This sounds good to me.

Reply:
@Geod24, Move to  source/agora/test/GossipProtocol.d